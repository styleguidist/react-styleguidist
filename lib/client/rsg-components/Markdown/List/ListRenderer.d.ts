import React from 'react';
import { JssInjectedProps } from 'rsg-components/Styled';
interface ListProps extends JssInjectedProps {
    ordered?: boolean;
    children: React.ReactNode;
}
export declare const ListRenderer: React.FunctionComponent<ListProps>;
declare const _default: React.ComponentType<Pick<ListProps, "children" | "ordered">>;
export default _default;
