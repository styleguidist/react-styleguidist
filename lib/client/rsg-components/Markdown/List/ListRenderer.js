import "core-js/modules/es.array.map";
import React, { cloneElement, Children } from 'react';
import PropTypes from 'prop-types';
import cx from 'clsx';
import Styled from 'rsg-components/Styled';

var styles = function styles(_ref) {
  var space = _ref.space,
      color = _ref.color,
      fontFamily = _ref.fontFamily;
  return {
    list: {
      marginTop: 0,
      marginBottom: space[2],
      paddingLeft: space[3],
      fontSize: 'inherit'
    },
    ordered: {
      listStyleType: 'decimal'
    },
    li: {
      color: color.base,
      fontFamily: fontFamily.base,
      fontSize: 'inherit',
      lineHeight: 1.5,
      listStyleType: 'inherit'
    }
  };
};

export var ListRenderer = function ListRenderer(_ref2) {
  var classes = _ref2.classes,
      ordered = _ref2.ordered,
      children = _ref2.children;
  var Tag = ordered ? 'ol' : 'ul';
  var classNames = cx(classes.list, ordered && classes.ordered);
  return /*#__PURE__*/React.createElement(Tag, {
    className: classNames
  }, Children.map(children, function (li) {
    return React.isValidElement(li) ? cloneElement(li, {
      className: classes.li
    }) : li;
  }));
};
ListRenderer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
  ordered: PropTypes.bool,
  children: PropTypes.node.isRequired
};
ListRenderer.defaultProps = {
  ordered: false
};
export default Styled(styles)(ListRenderer);