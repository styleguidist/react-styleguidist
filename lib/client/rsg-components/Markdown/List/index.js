export { default } from 'rsg-components/Markdown/List/ListRenderer';