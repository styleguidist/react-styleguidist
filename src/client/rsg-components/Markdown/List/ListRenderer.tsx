import React, { cloneElement, Children } from 'react';
import PropTypes from 'prop-types';
import cx from 'clsx';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';
import * as Rsg from '../../../../typings';

const styles = ({ space, color, fontFamily }: Rsg.Theme) => ({
	list: {
		marginTop: 0,
		marginBottom: space[2],
		paddingLeft: space[3],
		fontSize: 'inherit',
	},
	ordered: {
		listStyleType: 'decimal',
	},
	li: {
		color: color.base,
		fontFamily: fontFamily.base,
		fontSize: 'inherit',
		lineHeight: 1.5,
		listStyleType: 'inherit',
	},
});

interface ListProps extends JssInjectedProps {
	ordered?: boolean;
	children: React.ReactNode;
}

export const ListRenderer: React.FunctionComponent<ListProps> = ({
	classes,
	ordered,
	children,
}) => {
	const Tag = ordered ? 'ol' : 'ul';

	const classNames = cx(classes.list, ordered && classes.ordered);

	return (
		<Tag className={classNames}>
			{Children.map(children, (li) =>
				React.isValidElement(li) ? cloneElement(li, { className: classes.li }) : li
			)}
		</Tag>
	);
};
ListRenderer.propTypes = {
	classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
	ordered: PropTypes.bool,
	children: PropTypes.any.isRequired,
};
ListRenderer.defaultProps = {
	ordered: false,
};

export default Styled<ListProps>(styles)(ListRenderer);
