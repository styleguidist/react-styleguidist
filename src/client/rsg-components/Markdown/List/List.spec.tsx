import React from 'react';
import renderer from 'react-test-renderer';

import List from './index';

describe('Markdown List', () => {
	it('should render an unordered list', () => {
		const actual = renderer.create(
			<List>
				<li>First</li>
				<li>Second</li>
			</List>
		);

		expect(actual.toJSON()).toMatchSnapshot();
	});

	it('should render an ordered list', () => {
		const actual = renderer.create(
			<List ordered>
				<li>First</li>
				<li>Second</li>
			</List>
		);

		expect(actual.toJSON()).toMatchSnapshot();
	});
});
