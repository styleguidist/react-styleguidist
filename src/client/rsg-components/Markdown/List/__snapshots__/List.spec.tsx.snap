// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown List should render an ordered list 1`] = `
<ol
  className="rsg--list-0 rsg--ordered-1"
>
  <li
    className="rsg--li-2"
  >
    First
  </li>
  <li
    className="rsg--li-2"
  >
    Second
  </li>
</ol>
`;

exports[`Markdown List should render an unordered list 1`] = `
<ul
  className="rsg--list-0"
>
  <li
    className="rsg--li-2"
  >
    First
  </li>
  <li
    className="rsg--li-2"
  >
    Second
  </li>
</ul>
`;
